### PR TITLE
bintray: add files to downloads list (ports #950)

### PR DESCRIPTION
**Pull request by svenfuchs**

### CODE REVIEW

File: lib/dpl/providers/bintray.rb [hunk 1]
```diff
@@ -112,19 +115,41 @@ def publish_version
...
+      def update_files
+        files.each do |file|
+          info :upload_file, source: file.source, target: file.target
+          update_file(file)
+        end
+      end
+
+      def update_file(file)
+        retrying(max: 10, pause: 5) do
+          put(path(:file_metadata, target: file.target), list_in_downloads: true)
```
Comment (loicalbertin) [inline lib/dpl/providers/bintray.rb]: Why does `list_in_downloads` is always `true`? And I think that the request body should be converted into json.
I was expecting something like this:

```suggestion
          put(path(:file_metadata, target: file.target), {list_in_downloads: file.download}.to_json)
```

**Comment by loicalbertin** [comment]
I found several issues with this PR.
I fixed them into #1074.

**Comment by svenfuchs** [comment]
@loicalbertin thanks a lot for your work on this!

https://travis-ci.org/loicalbertin/travis-dpl-bintray-test/builds/573880501#L285 looks to me as if this could be merged as soon as i have resolved the merge conflicts on this branch?

i am assuming the uploaded files actually end up being listed on the downloads list?

**Comment by loicalbertin** [comment]
> i am assuming the uploaded files actually end up being listed on the downloads list?

Yes you can check it directly on [bintray overview for this release](https://bintray.com/beta/#/loicalbertin/dpl-test/auto-upload/0.6?tab=overview) while other files appear under [bintray files view](https://bintray.com/beta/#/loicalbertin/dpl-test/auto-upload/0.6?tab=files)

**Comment by svenfuchs** [comment]
@loicalbertin nice! 

ok, so i'll assume we can merge this then, once the build is green.

**Comment by loicalbertin** [comment]
Just a question with this new version of dpl where is the documentation for the bintray release descriptor file (I think it was previously in the README)?

**Comment by svenfuchs** [comment]
@loicalbertin good question, i guess i have overlooked that.

is there a page on the bintray documentation that we can link to, maybe? that way we can make sure our content does not get out of date, and cl has support for adding links to options

**Comment by svenfuchs** [comment]
wow, you're right, it's here: https://github.com/travis-ci/dpl/blob/v1/README.md#bintray

that's a lot of stuff to put into the help output of a command. i think it would be good to be able to link to bintray's documentation, and possibly add some of this to the provider's `description` on the class body. or just move all this to the Travis CI end user docs?

**Comment by loicalbertin** [comment]
This descriptor is dpl-specific so Travis CI end user docs is probably the right place to store it. [In fact it is already present](https://docs.travis-ci.com/user/deployment/bintray/#descriptor-file-example) the fact is I don't know which branch to use as this dpl v2 is not live yet :thinking:

**Comment by svenfuchs** [comment]
@loicalbertin we've sort of pushed documentation updates back a little, and only kept track of these on an internal todo list. 

we are considering branching the whole `deployments` section into two, as both versions might need to be available for a while, but we didn't want to get started on that until we know how many changes here actually will be.

if you'd like to make a PR for this, just branch off of `master`, and mention it's not supposed to be merged until v2 is out. we can then take care of any conflicts later.

**Comment by svenfuchs** [comment]
also, thanks for all the help on this @loicalbertin! ❤️

**Comment by loicalbertin** [comment]
Thank you for taking time to work on this.
I created the documentation PR travis-ci/docs-travis-ci-com#2488

**Comment by svenfuchs** [comment]
awesome, i have added this to our todo list 👍 ❤️

**Pull request merged**